src/Interpreters/Context.h

```diff
@@ -201,6 +214,9 @@ class Context
     using SampleBlockCache = std::unordered_map<std::string, Block>;
     mutable SampleBlockCache sample_block_cache;
 
+    std::shared_ptr<Fingerprints> fingerprints; /// set of parts' fingerprints, is used for query parts deduplication
```

Comment:
I'm confused by these names. As far as I understand the first set of fingerprints (aka `fingerprints` are all fingerprints participating in a query, the second one (aka `excluded_fingerprints`) is a set of parts fingerprints which the current server shouldn't read from the disk and send to the initiator?

Reply:
naming is hard, but you got it right.